sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -276,8 +290,39 @@ export class ApplicationService extends RecordDataModelService<Application> {
       application.currentAssessment.updatedAt = now;
 
       await applicationRepository.save([application, newApplication]);
+      // Create a notification if the application is edited for the 5th time.
+      const applicationsCount = await applicationRepository.count({
+        where: { applicationNumber: newApplication.applicationNumber },
+      });
+      if (
+        applicationsCount ===
+        APPLICATION_EDIT_COUNT_TO_SEND_NOTIFICATION + 1
+      ) {
+        const student = await this.studentRepo.findOne({
+          select: {
+            id: true,
+            birthDate: true,
+            user: { id: true, firstName: true, lastName: true, email: true },
+          },
+          relations: {
+            user: true,
+          },
+          where: { id: application.studentId },
+        });
+        const ministryNotification: ApplicationExceptionRequestNotificationForMinistry =
+          {
+            givenNames: student.user.firstName,
+            lastName: student.user.lastName,
+            email: student.user.email,
+            dob: student.birthDate,
+            applicationNumber: application.applicationNumber,
+          };
+        await this.notificationActionService.saveApplicationEditedFifthTimeNotificationForMinistry(
```

Comment:
All notification-related logic can be moved to a private method for better readability.
The `submitApplication` is already too long.